crates/pathfinder/src/state/sync/l2.rs

```diff
@@ -358,3 +371,1229 @@ async fn download_and_compress_contract(
+                },
+            };
+            static ref STATE_UPDATE2_V2: reply::StateUpdate = reply::StateUpdate {
+                new_root: *GLOBAL_ROOT2_V2,
+                old_root: *GLOBAL_ROOT1_V2,
+                state_diff: reply::state_update::StateDiff {
+                    deployed_contracts: vec![],
+                    storage_diffs: HashMap::new(),
+                },
+            };
+            static ref STATE_UPDATE3: reply::StateUpdate = reply::StateUpdate {
+                new_root: *GLOBAL_ROOT3,
+                old_root: *GLOBAL_ROOT2,
+                state_diff: reply::state_update::StateDiff {
+                    deployed_contracts: vec![],
+                    storage_diffs: HashMap::new(),
+                },
+            };
+
+            static ref EXPECTED_STATE_UPDATE0: state_update::StateUpdate = state_update::StateUpdate {
+                contract_updates: vec![state_update::ContractUpdate {
+                    address: *CONTRACT0_ADDR,
+                    storage_updates: vec![
+                        state_update::StorageUpdate {
+                            address: *STORAGE_KEY0,
+                            value: *STORAGE_VAL0,
+                        }
+                    ]
+                }],
+                deployed_contracts: vec![
+                    state::sync::DeployedContract {
+                        address: *CONTRACT0_ADDR,
+                        hash: *CONTRACT0_HASH,
+                        call_data: vec![],
+                }],
+            };
+            static ref EXPECTED_STATE_UPDATE1: state_update::StateUpdate = state_update::StateUpdate {
+                contract_updates: vec![
+                    state_update::ContractUpdate {
+                        address: *CONTRACT0_ADDR,
+                        storage_updates: vec![
+                            state_update::StorageUpdate {
+                                address: *STORAGE_KEY0,
+                                value: *STORAGE_VAL0_V2,
+                            }
+                        ]
+                    },
+                    state_update::ContractUpdate {
+                        address: *CONTRACT1_ADDR,
+                        storage_updates: vec![
+                            state_update::StorageUpdate {
+                                address: *STORAGE_KEY1,
+                                value: *STORAGE_VAL1,
+                            }
+                        ]
+                    }
+                ],
+                deployed_contracts: vec![
+                    state::sync::DeployedContract {
+                        address: *CONTRACT1_ADDR,
+                        hash: *CONTRACT1_HASH,
+                        call_data: vec![],
+                }],
+            };
+        }
+
+        /// Convenience wrapper
+        fn expect_block(
+            mock: &mut MockClientApi,
+            seq: &mut mockall::Sequence,
```

Comment:
Did it cutoff the `mockall::Sequence[r]`?

Reply:
`seq` is [an expected sequence of mock calls](https://docs.rs/mockall/latest/mockall/struct.Sequence.html), has nothing to do with the sequencer::Client if I understood the question correctly :thinking: